crates/rune/src/any.rs

```diff
@@ -18,7 +22,32 @@ pub use rune_macros::Any;
 ///     name: String,
 /// }
 /// ```
-pub trait Any: Named + MetaFieldMode + std::any::Any {
+#[cfg(not(feature = "dynamic_fields"))]
```

Comment:
Hm, avoid feature gating two incompatible traits of `Any`. What can happen is that you have coherence issues under a scenario such as:

Crate `a` defines a type that implements `Any` without activating the `dynamic_fields` feature, so they leave `FIELD_MODE` undefined.

Crate `b` defines a type that implements `Any` while activating the `dynamic_fields` feature, so they define `FIELD_MODE`.

Crate `c` depends on `a` and `b`, so the `dynamic_fields` feature is activated, causing `a` to fail to build.

Instead:
* Unconditionally include `FIELD_MODE` (and possibly rename to `DYNAMIC_FIELD_MODE`).
* Feature gate the other variants of `DynamicFieldMode` which are expected to cause alternate behaviors (`First`, `Last`, and `Only`).
* Mark `DynamicFieldMode` as `#[non_exhaustive]`.